syllabus.md

```diff
@@ -0,0 +1,32 @@
+# Syllabus
+
+The following topics relate to code quality.
+
+- [ ] SW Interfaces
+- [x] The build pattern
+- [ ] Anonymous function (lambda)
+  - Closure
+- [ ] The SOLID principles
+  - [ ] Single responsibility principle
+  - [ ] Open-Closed principle
+  - [ ] Liskov Substitution principle
+  - [ ] Interface segregation principle
+  - [ ] Dependency inversion principle
+- [ ] Immutable functions
```

Comment:
Could you please elaborate on the meaning of "Immutable functions"?

Reply:
Functions that do not change the input.